WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -751,24 +751,15 @@ private void SetFees()
 				}
 
 				long all = selectedCoins.Sum(x => x.Amount);
-				if (IsMax)
-				{
-					if (all != 0)
-					{
-						FeePercentage = 100 * (decimal)EstimatedBtcFee.Satoshi / all;
-					}
-					else
-					{
-						FeePercentage = 0;
-					}
-				}
-				else
-				{
-					if (Money.TryParse(AmountText.TrimStart('~', ' '), out Money amount) && amount.Satoshi != 0)
-					{
-						FeePercentage = 100 * (decimal)EstimatedBtcFee.Satoshi / amount.Satoshi;
-					}
-				}
+				long theAmount = IsMax
+					? all
+					: Money.TryParse(AmountText.TrimStart('~', ' '), out Money value)
+						? value.Satoshi
+						: 0;
```

Comment:
It looks much better. What do you think about doing it as below? IMO is clearer.

```c#
				long theAmount = (IsMax, Money.TryParse(AmountText.TrimStart('~', ' '), out Money value)) switch 
				{
					(true, _) => all,
					(false, true) => value.Satoshi,
					(false, false) => 0
				};

```

Reply:
Done